packages/strapi-design-system/src/SubNav/SubNavHeader.js

```diff
@@ -0,0 +1,135 @@
+import React, { useState, useRef, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import Search from '@strapi/icons/Search';
+import CloseAlertIcon from '@strapi/icons/CloseAlertIcon';
+import SearchIcon from '@strapi/icons/SearchIcon';
+import { Row } from '../Row';
+import { H2 } from '../Text';
+import { IconButton } from '../IconButton';
+import { Box } from '../Box';
+import { FieldInput, FieldAction } from '../Field';
+import { Divider } from '../Divider';
+import { useId } from '../helpers/useId';
+import { usePrevious } from '../helpers/usePrevious';
+import { KeyboardKeys } from '../helpers/keyboardKeys';
+
+const Searchbar = styled(FieldInput)`
+  height: ${32 / 16}rem;
+`;
+const CloseIconWrapper = styled(Row)`
+  font-size: 0.5rem;
+  svg path {
+    fill: ${({ theme }) => theme.colors.neutral400};
+  }
+`;
+const SearchIconWrapper = styled(Row)`
+  font-size: 0.8rem;
+  svg path {
+    fill: ${({ theme, focused }) => (focused ? theme.colors.primary600 : theme.colors.neutral800)};
+  }
+`;
+const CustomDivider = styled(Divider)`
+  width: ${24 / 16}rem;
+`;
+
+export const SubNavHeader = ({ label, searchLabel, searchable, onChange, value, onClear }) => {
+  const [isSearchOpen, setSearchOpen] = useState(false);
+  const previousSearchOpenValue = usePrevious(isSearchOpen);
+  const clearButonId = useId('subnav-searchbar-clear');
+  const searchRef = useRef();
+  const searchButtonRef = useRef();
+  const isCompleting = value.length > 0;
+
+  useEffect(() => {
+    if (isSearchOpen && searchRef.current) {
+      searchRef.current.focus();
+    }
+    if (previousSearchOpenValue && !isSearchOpen && searchButtonRef.current) {
+      searchButtonRef.current.focus();
+    }
+  }, [isSearchOpen]);
+
+  const toggleSearch = (e) => {
+    setSearchOpen((isOpen) => !isOpen);
+  };
+
+  const handleClear = (e) => {
+    onClear(e);
+    searchRef.current.focus();
+  };
+
+  const handleBlur = (e) => {
+    if (e.relatedTarget?.id !== clearButonId) {
+      setSearchOpen(false);
+    }
+  };
+
+  const handleKeyDown = (e) => {
+    if (e.key === KeyboardKeys.ESCAPE) {
+      setSearchOpen(false);
+    }
+  };
+
+  if (isSearchOpen) {
+    return (
+      <Box paddingLeft={4} paddingTop={5} paddingBottom={2} paddingRight={4}>
+        <Searchbar
```

Comment:
So the list will be filtered by the parent right?

Can you add the search logic to the story maybe using matchsorter?

Reply:
Don't you think that the search should be handled by the component itself?

Reply:
I think this should be handled in the implementation, as the SubNavHeader is a different component from the list and is not related to it at all. If we want to add the search logic to the component, we will have to modify the API so that the SubNavHeader can render the list or provide the filtered list to other components.

Reply:
So does it mean that we will always have to implement the search logic for this component?
Maybe you could provide a hook example that we will later implement in Strapi for this in the CRA.

What do you think?

Reply:
I will add this in the CRA with pleasure in another PR if you're ok with that?

Reply:
Sure, I'll look forward for this PR!